prober/derp.go

```diff
@@ -282,40 +344,96 @@ func derpProbeNodePair(ctx context.Context, dm *tailcfg.DERPMap, from, to *tailc
 		time.Sleep(100 * time.Millisecond) // pretty arbitrary
 	}
 
-	latency, err = runDerpProbeNodePair(ctx, from, to, fromc, toc)
+	return runDerpProbeNodePair(ctx, from, to, fromc, toc, size)
+}
+
+// derpProbeBandwidth sends a small packet between two local DERP clients
+// connected to two DERP servers.
+func derpProbeNodePair(ctx context.Context, dm *tailcfg.DERPMap, from, to *tailcfg.DERPNode) (err error) {
+	fromc, err := newConn(ctx, dm, from, true)
 	if err != nil {
+		return err
+	}
+	defer fromc.Close()
+	toc, err := newConn(ctx, dm, to, true)
+	if err != nil {
+		return err
+	}
+	defer toc.Close()
+
+	// Wait a bit for from's node to hear about to existing on the
+	// other node in the region, in the case where the two nodes
+	// are different.
+	if from.Name != to.Name {
+		time.Sleep(100 * time.Millisecond) // pretty arbitrary
```

Comment:
Does it matter what happens if 100ms isn't enough? Will that lead to a prober failure, or will it just retry later and only complain if it keeps happening? (Maybe that would be obvious in a wider context, but I don't see how this can possibly not be flaky)